src/linear_algebra/finite_dimensional.lean

```diff
@@ -1759,6 +1756,50 @@ theorem subalgebra.dim_eq_one_iff {S : subalgebra F E} : module.rank F S = 1 ↔
 theorem subalgebra.finrank_eq_one_iff {S : subalgebra F E} : finrank F S = 1 ↔ S = ⊥ :=
 ⟨subalgebra.eq_bot_of_finrank_one, subalgebra.finrank_eq_one_of_eq_bot⟩
 
+lemma subalgebra.bot_eq_top_iff_dim_eq_one :
+  module.rank F E = 1 ↔ (⊥ : subalgebra F E) = ⊤ :=
+begin
+  rw [← dim_top, ← subalgebra_top_dim_eq_submodule_top_dim],
+  split,
+  { exact λ h, (subalgebra.eq_bot_of_dim_one h).symm, },
+  { intro h,
+    haveI : finite_dimensional F (⊤ : subalgebra F E),
+    { rw ← h, exact subalgebra.finite_dimensional_bot, },
+    rw_mod_cast [← finite_dimensional.finrank_eq_dim, subalgebra.finrank_eq_one_iff],
+    exact h.symm, }
+end
+
+@[simp]
+lemma subalgebra.bot_eq_top_of_dim_eq_one (h : module.rank F E = 1) : (⊥ : subalgebra F E) = ⊤ :=
+subalgebra.bot_eq_top_iff_dim_eq_one.mp h
+
+def subalgebra.is_simple_order_of_finrank (hr : finrank F E = 2) :
+  is_simple_order (subalgebra F E) :=
```

Comment:
Hmm, I just realized this section assumes `E` is also a field, which seems too strong; `[ring E]` should work (some may need `[nontrivial E]`). If `E` is also a field, we can get the same conclusion assuming that `finrank F E` is a prime.

Reply:
Well, I guess that would be for another PR ;)